src/cloudai/systems/slurm/strategy/slurm_command_gen_strategy.py

```diff
@@ -120,14 +139,89 @@ def job_name(self, job_name_prefix: str) -> str:
             job_name = f"{self.system.account}-{job_name_prefix}.{datetime.now().strftime('%Y%m%d_%H%M%S')}"
         return job_name
 
-    def generate_srun_command(
+    def gen_prologue(self, prologue: TestScenario, base_output_path: Path) -> str:
+        """
+        Generate the prologue command by running all tests defined in the prologue test scenario.
+
+        Args:
+            prologue (TestScenario): The prologue test scenario containing the tests to be run.
+            base_output_path (Path): The base output directory path for storing prologue outputs.
+
+        Returns:
+            str: A string with all the Slurm srun commands generated for the prologue.
+        """
+        if not prologue.test_runs:
```

Comment:
Is it possible? Pydantic model defines list of test runs to be non-empty.

Reply:
Do you mean that this is not allowed by Pydantic?

Reply:
Yeah, when a TestScenario object is created it is verified against our schema which requires at least one test case to be defined.

But if you saw such cases, we need to check why it happened and keep this guarding check, of course.